Test submission by João Souza

joaosouza1 opened the pull request:
Thank you for reviewing my pull request. I'm looking forward to see your comments and questions. Please read the updated `README` for my comments. Feel free to reach me at: joaomarcelo9393@gmail.com

mat-sf reviewed (commented):
Hi João Souza. First of all, I would like to thank you for spending time on this submission and that very well-described readme, really appreciate that.

Besides the questions in the PR I have two more general ones:
1. Do you have any Graphql experience?
2. Do you have any experience in writing e2e tests? What do you prefer to use?

Have a nice day!

joaosouza1 reviewed (commented):
Hi @mat-sf! Thank you for the review. To answer your questions:

1. **Do you have any Graphql experience?**
A little, from a test project, not in production. I know what it is, I know you can use [GraphiQL](https://github.com/graphql/graphiql) for documentation and that you can fetch data with [Apollo](https://www.apollographql.com/docs/react/) or [SWR](https://swr.vercel.app/docs/data-fetching#graphql).

2. **Do you have any experience in writing e2e tests? What do you prefer to use?**
Yes, I can write e2e tests. I know that Cypress is the go-to tool currently, but I don't have any experience with it, although I'm already looking forward to learn it. At my current job, they have a Ruby backend, so I'm used to writing and running e2e tests with RSpec, Capybara and Selenium. In other words, I'm familiar with e2e tests and I can learn a different tool.

I'm available to answer anything else you'd like to know. Cheers!

joaosouza1 commented:
@mat-sf Here's an update that makes filters accessible by keyboard and screen readers: https://github.com/Superformula/frontend-test/pull/46/commits/f0d7966a9bdff28a24ea5a514cbd6844e46c1ce6

Regarding this question: https://github.com/Superformula/frontend-test/pull/46#discussion_r755618009

joaosouza1 commented:
@mat-sf Tonight on my free time I fixed many problems from my previous submission. You can find details in the last 4 commit messages. In summary:

- I unified TS transpiling in Babel (previously, Webpack did it for dev environment, and Babel did it for test environment).
- I fixed a console warning regarding the `primary` prop on Link (the CTA element passed it through).
- I fixed the Restaurant Card and the Restaurant Grid in Storybook (they missed the MemoryRouter).
- I updated the RestaurantGrid test suite, now all tests pass.

joaosouza1 commented:
😅 ...And as one last touch, I set up the production build. Details in the commit message. I believe this is the last bit of post-submission improvement I'll be adding. Cheers!
